From 8005f4ee748b1fe324b3f234a2defe0dd557611b Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Mon, 12 Mar 2018 15:41:26 +0000
Subject: [PATCH] OS-6412 ship initial bhyve bits Reviewed by: Mike Gerdts
 <mike.gerdts@joyent.com> Reviewed by: Jerry Jelinek
 <jerry.jelinek@joyent.com> Approved by: Jerry Jelinek
 <jerry.jelinek@joyent.com>

---
 manifest                                      | 14 +++++
 .../pkg/manifests/developer-build-onbld.mf    |  1 +
 usr/src/pkg/manifests/system-bhyve.mf         | 62 +++++++++++++++++++
 3 files changed, 77 insertions(+)
 create mode 100644 usr/src/pkg/manifests/system-bhyve.mf

diff --git a/manifest b/manifest
index 920959189d..5262bdb261 100644
--- a/manifest
+++ b/manifest
@@ -1307,6 +1307,8 @@ f lib/amd64/libuuid.so.1 0755 root bin
 s lib/amd64/libuuid.so=libuuid.so.1
 f lib/amd64/libuutil.so.1 0755 root bin
 s lib/amd64/libuutil.so=libuutil.so.1
+f lib/amd64/libvmmapi.so.1 0755 root bin
+s lib/amd64/libvmmapi.so=libvmmapi.so.1
 f lib/amd64/libw.so.1 0755 root bin
 s lib/amd64/libw.so=libw.so.1
 f lib/amd64/libxnet.so.1 0755 root bin
@@ -4574,6 +4576,7 @@ f usr/kernel/drv/amd64/lxautofs 0755 root sys
 f usr/kernel/drv/amd64/nsmb 0755 root sys
 f usr/kernel/drv/amd64/pm 0755 root sys
 f usr/kernel/drv/amd64/pool 0755 root sys
+f usr/kernel/drv/amd64/ppt 0755 root sys
 f usr/kernel/drv/amd64/ptm 0755 root sys
 f usr/kernel/drv/amd64/pts 0755 root sys
 f usr/kernel/drv/amd64/signalfd 0755 root sys
@@ -4581,6 +4584,8 @@ f usr/kernel/drv/amd64/smbsrv 0755 root sys
 f usr/kernel/drv/amd64/sppp 0755 root sys
 f usr/kernel/drv/amd64/sppptun 0755 root sys
 f usr/kernel/drv/amd64/timerfd 0755 root sys
+f usr/kernel/drv/amd64/viona 0755 root sys
+f usr/kernel/drv/amd64/vmm 0755 root sys
 f usr/kernel/drv/amd64/zcons 0755 root sys
 f usr/kernel/drv/amd64/zfd 0755 root sys
 f usr/kernel/drv/bpf.conf 0644 root sys
@@ -4599,6 +4604,7 @@ f usr/kernel/drv/lx_systrace.conf 0644 root sys
 f usr/kernel/drv/nsmb.conf 0644 root sys
 f usr/kernel/drv/pm.conf 0644 root sys
 f usr/kernel/drv/pool.conf 0644 root sys
+f usr/kernel/drv/ppt.conf 0644 root sys
 f usr/kernel/drv/ptm.conf 0644 root sys
 f usr/kernel/drv/pts.conf 0644 root sys
 f usr/kernel/drv/signalfd.conf 0644 root sys
@@ -4606,6 +4612,8 @@ f usr/kernel/drv/smbsrv.conf 0644 root sys
 f usr/kernel/drv/sppp.conf 0644 root sys
 f usr/kernel/drv/sppptun.conf 0644 root sys
 f usr/kernel/drv/timerfd.conf 0644 root sys
+f usr/kernel/drv/viona.conf 0644 root sys
+f usr/kernel/drv/vmm.conf 0644 root sys
 d usr/kernel/fs 0755 root sys
 d usr/kernel/fs/amd64 0755 root sys
 f usr/kernel/fs/amd64/fdfs 0755 root sys
@@ -5099,6 +5107,7 @@ f usr/lib/brand/bhyve/uninstall 0555 root sys
 f usr/lib/brand/bhyve/config.xml 0444 root sys
 f usr/lib/brand/bhyve/platform.xml 0444 root sys
 f usr/lib/brand/bhyve/statechange 0555 root sys
+h usr/lib/brand/bhyve/zhyve=usr/sbin/amd64/bhyve
 d usr/lib/brand/lx 0755 root bin
 s usr/lib/brand/lx/64=amd64
 d usr/lib/brand/lx/amd64 0755 root bin
@@ -9601,6 +9610,7 @@ f usr/lib/mdb/kvm/amd64/sv.so 0555 root sys
 f usr/lib/mdb/kvm/amd64/ufs.so 0555 root sys
 f usr/lib/mdb/kvm/amd64/uhci.so 0555 root sys
 f usr/lib/mdb/kvm/amd64/usba.so 0555 root sys
+f usr/lib/mdb/kvm/amd64/vmm.so 0555 root sys
 f usr/lib/mdb/kvm/amd64/xhci.so 0555 root sys
 f usr/lib/mdb/kvm/amd64/zfs.so 0555 root sys
 d usr/lib/mdb/proc 0755 root sys
@@ -10263,6 +10273,8 @@ f usr/sbin/acpixtract 0555 root bin
 f usr/sbin/add_drv 0555 root bin
 f usr/sbin/allocate 4555 root bin
 d usr/sbin/amd64 0755 root bin
+f usr/sbin/amd64/bhyve 0555 root sys
+f usr/sbin/amd64/bhyvectl 0555 root sys
 f usr/sbin/amd64/dtrace 0555 root bin
 f usr/sbin/amd64/intrstat 0555 root bin
 f usr/sbin/amd64/ipf 0555 root bin
@@ -10284,6 +10296,8 @@ f usr/sbin/auditstat 0555 root bin
 h usr/sbin/audlinks=usr/sbin/devfsadm
 s usr/sbin/automount=../lib/fs/autofs/automount
 s usr/sbin/autopush=../../sbin/autopush
+h usr/sbin/bhyve=usr/lib/isaexec
+h usr/sbin/bhyvectl=usr/lib/isaexec
 f usr/sbin/cfgadm 0555 root bin
 f usr/sbin/check-hostname 0555 root mail
 f usr/sbin/check-permissions 0555 root mail
diff --git a/usr/src/pkg/manifests/developer-build-onbld.mf b/usr/src/pkg/manifests/developer-build-onbld.mf
index 4ce63a7dcb..f7e60d3db3 100644
--- a/usr/src/pkg/manifests/developer-build-onbld.mf
+++ b/usr/src/pkg/manifests/developer-build-onbld.mf
@@ -101,6 +101,7 @@ file path=opt/onbld/bin/find_elf mode=0555
 file path=opt/onbld/bin/findcrypto mode=0555
 file path=opt/onbld/bin/flg.flp mode=0555
 file path=opt/onbld/bin/genoffsets mode=0555
+file path=opt/onbld/bin/gensetdefs mode=0555
 file path=opt/onbld/bin/git-pbchk mode=0555
 file path=opt/onbld/bin/hdrchk mode=0555
 file path=opt/onbld/bin/interface_check mode=0555
diff --git a/usr/src/pkg/manifests/system-bhyve.mf b/usr/src/pkg/manifests/system-bhyve.mf
new file mode 100644
index 0000000000..3faf955e20
--- /dev/null
+++ b/usr/src/pkg/manifests/system-bhyve.mf
@@ -0,0 +1,62 @@
+#
+# CDDL HEADER START
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2018 Joyent, Inc.
+#
+
+#
+# The default for payload-bearing actions in this package is to appear in the
+# global zone only.  See the include file for greater detail, as well as
+# information about overriding the defaults.
+#
+<include global_zone_only_component>
+set name=pkg.fmri value=pkg:/system/bhyve@$(PKGVERS)
+set name=pkg.description value="BSD hypervisor"
+set name=pkg.summary value="BSD hypervisor"
+set name=info.classification \
+    value=org.opensolaris.category.2008:System/Virtualization
+set name=variant.arch value=i386
+dir path=kernel group=sys
+dir path=kernel/kmdb group=sys
+dir path=kernel/kmdb/$(ARCH64) group=sys
+dir path=lib group=bin
+dir path=lib/$(ARCH64) group=bin
+dir path=usr group=sys
+dir path=usr/kernel/drv group=sys
+dir path=usr/kernel/drv/$(ARCH64) group=sys
+dir path=usr/lib group=bin
+dir path=usr/lib/mdb group=sys
+dir path=usr/lib/mdb/kvm group=sys
+dir path=usr/lib/mdb/kvm/$(ARCH64) group=sys
+dir path=usr/sbin
+driver name=ppt
+driver name=viona
+driver name=vmm
+file path=kernel/kmdb/$(ARCH64)/vmm mode=0555
+file path=lib/$(ARCH64)/libvmmapi.so.1
+file path=usr/kernel/drv/$(ARCH64)/ppt
+file path=usr/kernel/drv/$(ARCH64)/viona
+file path=usr/kernel/drv/$(ARCH64)/vmm
+file path=usr/kernel/drv/ppt.conf
+file path=usr/kernel/drv/viona.conf
+file path=usr/kernel/drv/vmm.conf
+file path=usr/lib/mdb/kvm/$(ARCH64)/vmm.so mode=0555
+file path=usr/sbin/$(ARCH64)/bhyve mode=0555
+file path=usr/sbin/$(ARCH64)/bhyvectl mode=0555
+hardlink path=usr/sbin/bhyve target=../lib/isaexec
+hardlink path=usr/sbin/bhyvectl target=../lib/isaexec
+license lic_CDDL license=lic_CDDL
+link path=lib/$(ARCH64)/libvmmapi.so target=./libvmmapi.so.1
-- 
2.21.0

